src/resolvers/Mutation/createEvent.ts

```diff
@@ -119,39 +119,35 @@ export const createEvent: MutationResolvers["createEvent"] = async (
     );
   }
 
-  // Creates new event.
-  const createdEvent = await Event.create({
-    ...args.data,
-    creator: currentUser._id,
-    admins: [currentUser._id],
-    organization: organization._id,
-  });
+  let createdEvent;
 
-  if (createdEvent !== null) {
-    await cacheEvents([createdEvent]);
-  }
+  if (
+    !args.data?.recurring ||
+    (args.data?.recurring && args.data?.recurrance == "ONCE")
+  ) {
+    createdEvent = await Event.create({
+      ...args.data,
+      creator: currentUser._id,
+      admins: [currentUser._id],
+      organization: organization._id,
+    });
+
+    if (createdEvent !== null) {
+      await cacheEvents([createdEvent]);
+    }
 
-  await EventAttendee.create({
-    userId: currentUser._id.toString(),
-    eventId: createdEvent._id,
-  });
+    await associateEventWithUser(currentUser, createdEvent);
```

Comment:
Let say your db call for event creation is success but this call fails. Then as per your current logic the partial implementation will be accepted. Please start a transaction and make sure everything is success or everything fails.